x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/ml/integration/MlRestTestStateCleaner.java

```diff
@@ -54,4 +55,12 @@ private void deleteAllTrainedModelIngestPipelines() throws IOException {
             }
         }
     }
+
+    private void waitForMlStatsIndexToInitialize() throws IOException {
+        ESRestTestCase.ensureHealth(".ml-stats-*", (request) -> {
```

Comment:
This should fix the failing tests? It looks like it's failing on 403s: 
```
{"error":{"root_cause":[{"type":"security_exception","reason":"action [cluster:monitor/health] is unauthorized for user [no_ml] with effective roles [minimal], this action is granted by the cluster privileges [monitor,manage,all]"}],"type":"security_exception","reason":"action [cluster:monitor/health] is unauthorized for user [no_ml] with effective roles [minimal], this action is granted by the cluster privileges [monitor,manage,all]"},"status":403}
```
```suggestion
        ESRestTestCase.ensureHealth(adminClient, ".ml-stats-*", (request) -> {
```